### PR TITLE
Add insert param template parsing in nested parts

### PR DESCRIPTION
**Pull request by minhaj10p**

### CODE REVIEW

File: types/oscal/catalog/catalog.go [hunk 1]
```diff
@@ -127,6 +127,28 @@ type Part struct {
...
+
+	part.Prose.ReplaceInsertParams(parameterID, parameterVal)
+	if len(part.Parts) == 0 {
+		return
+	}
+	for i := range part.Parts {
+		traverseParts(&part.Parts[i], parameterID, parameterVal)
+	}
+	return
+
```
Comment (asadullah-yousuf-10p) [inline types/oscal/catalog/catalog.go]: Remove extra line

File: types/oscal/catalog/catalog.go [hunk 2]
```diff
@@ -127,6 +127,28 @@ type Part struct {
...
+}
+func (part *Part) ModifyProse(parameterID, parameterVal string) {
```
Comment (asadullah-yousuf-10p) [inline types/oscal/catalog/catalog.go]: Add line break

File: types/oscal/catalog/catalog_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package catalog
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestModifyParts(t *testing.T) {
+
```
Comment (asadullah-yousuf-10p) [inline types/oscal/catalog/catalog_test.go]: Remove line break

**Comment by minhaj10p** [comment]
closed in reference to https://github.com/docker/oscalkit/pull/63

**Pull request closed without merging**